react/features/always-on-top/AlwaysOnTop.js

```diff
@@ -0,0 +1,295 @@
+
+        this._audioEnabledListener = this._audioEnabledListener.bind(this);
+        this._audioMutedListener = this._audioMutedListener.bind(this);
+        this._mouseMove = this._mouseMove.bind(this);
+        this._onMouseOver = this._onMouseOver.bind(this);
+        this._onMouseOut = this._onMouseOut.bind(this);
+        this._videoEnabledListener = this._videoEnabledListener.bind(this);
+        this._videoMutedListener = this._videoMutedListener.bind(this);
+    }
+
+    /**
+     * Handles audio enabled api events.
+     *
+     * @param {{ enabled: boolean }} status - The new enabled status.
+     * @returns {void}
+     */
+    _audioEnabledListener({ enabled }) {
+        this.setState({ audioEnabled: enabled });
+    }
+
+    /**
+     * Handles audio muted api events.
+     *
+     * @param {{ muted: boolean }} status - The new muted status.
+     * @returns {void}
+     */
+    _audioMutedListener({ muted }) {
+        this.setState({ audioMuted: muted });
+    }
+
+    /**
+     * Hides the toolbar after a timeout.
+     *
+     * @returns {void}
+     */
+    _hideToolbarAfterTimeout() {
+        setTimeout(() => {
+            if (this._hovered) {
+                this._hideToolbarAfterTimeout();
+
+                return;
+            }
+            this.setState({ visible: false });
+        }, TOOLBAR_TIMEOUT);
+    }
+
+    /**
+     * Handles mouse move events.
+     *
+     * @returns {void}
+     */
+    _mouseMove() {
+        if (!this.state.visible) {
+            this.setState({ visible: true });
+        }
+    }
+
+    /**
+     * Toolbar mouse over handler.
+     *
+     * @returns {void}
+     */
+    _onMouseOver() {
+        this._hovered = true;
+    }
+
+    /**
+     * Toolbar mouse out handler.
+     *
+     * @returns {void}
+     */
+    _onMouseOut() {
+        this._hovered = false;
+    }
+
+    /**
+     * Handles audio enabled api events.
+     *
+     * @param {{ enabled: boolean }} status - The new enabled status.
+     * @returns {void}
+     */
+    _videoEnabledListener({ enabled }) {
+        this.setState({ videoMuted: enabled });
+    }
+
+    /**
+     * Handles video muted api events.
+     *
+     * @param {{ muted: boolean }} status - The new muted status.
+     * @returns {void}
+     */
+    _videoMutedListener({ muted }) {
+        this.setState({ videoMuted: muted });
+    }
+
+    /**
+     * Sets mouse move listener and initial toolbar timeout.
+     *
+     * @inheritdoc
+     * @returns {void}
+     */
+    componentDidMount() {
+        api.on('audioMuteStatusChanged', this._audioMutedListener);
+        api.on('videoMuteStatusChanged', this._videoMutedListener);
+        api.on('audioEnabledStatusChanged', this._audioEnabledListener);
+        api.on('videoEnabledStatusChanged', this._videoEnabledListener);
+
+        Promise.all([
+            api.isAudioMuted(),
+            api.isVideoMuted(),
+            api.isAudioAvailable(),
+            api.isVideoAvailable()
```

Comment:
They are absolutely the same. Initially I was using enabled for everything because this was the naming in jitsi-meet. But after that when I was rebasing the branch everything was changed to available. Probably I forgot to rename the events. Will fix it!